torchio/transforms/preprocessing/intensity/projection.py

```diff
@@ -0,0 +1,158 @@
+from torchio.data.image import ScalarImage
+from ....data.subject import Subject
+from ...intensity_transform import IntensityTransform
+from typing import Optional
+from math import ceil
+
+
+class Projection(IntensityTransform):
+    """Project intensities along a given axis, possibly with sliding slabs.
+
+    Args:
+        axis: Possible inputs are ``'Left'``, ``'Right'``, ``'Anterior'``,
+                ``'Posterior'``, ``'Inferior'``, ``'Superior'``. Lower-case
+                versions and first letters are also valid, as only the first
+                letter will be used.
+        slab_thickness: Thickness of slab projections. In other words, the
+            number of voxels in the ``axis`` dimension to project across.
+            If ``None``, the projection will be done across the entire span of
+            the ``axis`` dimension (i.e. ``axis`` dimension will be reduced to
+            1).
+        stride: Number of voxels to stride along the ``axis`` dimension between
+            slab projections.
+        projection_type: Type of intensity projection. Possible inputs are
+            ``'max'`` (the default), ``'min'``, ``'mean'``, ``'median'``, or
+            ``'quantile'``. If ``'quantile'`` is used, ``q`` must also be
+            supplied.
+        q: Quantile to use for intensity projections. This argument is required
+            if ``projection_type`` is ``'quantile'`` and is silently ignored
+            otherwise.
+        full_slabs_only: Boolean. Should projections be done only for slabs
+            that are ``slab_thickness`` thick? Default is ``True``.
+            If ``False``, some slabs may not be ``slab_thickness`` thick
+            depending on the size of the image, slab thickness, and stride.
+
+    Example:
+        >>> import torchio as tio
+        >>> ct = tio.datasets.Slicer('CTChest').CT_chest
+        >>> axial_mips = tio.Projection("S", slab_thickness=20, stride=20)
+        >>> ct_t = axial_mips(ct)
+        >>> ct_t.plot()
+
+    .. plot::
+
+        import torchio as tio
+        sub = tio.datasets.Slicer('CTChest')
+        ct = sub.CT_chest
+        axial_mips = tio.Projection("S", slab_thickness=20, stride=20)
+        ct_mips = axial_mips(ct)
+        sub.add_image(ct_mips, 'MIP')
+        sub.plot()
+
+    """
+    def __init__(
+            self,
+            axis: str,
+            slab_thickness: Optional[int] = None,
+            stride: Optional[int] = 1,
+            projection_type: Optional[str] = 'max',
+            q: Optional[float] = None,
+            full_slabs_only: Optional[bool] = True,
+            **kwargs
+            ):
+        super().__init__(**kwargs)
+        self.args_names = (
+            'axis', 'slab_thickness', 'stride',
+            'projection_type', 'q', 'full_slabs_only'
+            )
+        self.axis = axis
+        self.slab_thickness = slab_thickness
+        self.stride = stride
+        self.projection_type = projection_type
+        self.q = q
```

Comment:
Can we use a more descriptive variable name? E.g. `quantile` (or `percentile`). By the way, if we do change the name of the argument to `percentile`, you can just divide by 100 here so the rest of the code stays the same.